app/models/speaker.rb

```diff
@@ -9,8 +9,15 @@ class Speaker < ActiveRecord::Base
 
   serialize :info, Hash
 
-  validates :user, :event, presence: true
+  validates :event, presence: true
   validates :bio, length: {maximum: 500}
+  validates :name, :email, presence: true
+
+  scope :with_program_session, -> { Speaker.where("program_session_id IS NOT NULL") }
```

Comment:
this scope seems more complicated than it needs to be.  Also, it reads very technically but what is the concept that it captures?   Would you talk to new organizers with this language?  

My first thought is that these are program speakers.  So current_event.speakers.in_program seems more natural.